File: String/Unique Length-3 Palindromic Subsequences/solution.ts
function countPalindromicSubsequence(s: string): number {
    let result = 0
    let first = new Map()
    let last = new Map()
    for(let i = 0; i < s.length; i++) {
        if(!first.has(s[i])) first.set(s[i], i)
        last.set(s[i], i)
    }
    for(let letter of first.keys()) {
        let set = new Set()
        for(let i = first.get(letter) + 1; i < last.get(letter); i++) {
            set.add(s[i])
        }
        result += set.size
    }
    return result
};

File: String/Unique Length-3 Palindromic Subsequences/solution.py
class Solution:
    def countPalindromicSubsequence(self, s: str) -> int:
        first = dict()
        last = dict()
        result = 0
        for i, c in enumerate(s):
            if c not in first: first[c] = i
            last[c] = i
        for c in first.keys():
            i, j = first[c], last[c]
            if i < j:
                result += len(set(s[i+1:j]))
        return result
